addons/barcodes/models/barcode_nomenclature.py

```diff
@@ -155,12 +144,12 @@ def parse_barcode(self, barcode):
 
             match = self.match_pattern(cur_barcode, rule['pattern'])
             if match['match']:
-                if rule['type'] == 'alias':
+                if rule['rule_type'] == 'alias':
                     barcode = rule['alias']
                     parsed_result['code'] = barcode
                 else:
                     parsed_result['encoding'] = rule['encoding']
-                    parsed_result['type'] = rule['type']
+                    parsed_result['rule_type'] = rule['rule_type']
```

Comment:
Now your parsed result sending key as rule_type, so you also need to do changes from where this parse_barcode is called otherwise it may throw traceback while accessing type from parsed result